nextstrain/cli/runner/managed_conda.py

```diff
@@ -0,0 +1,337 @@
+"""
+Run commands with access to a fully-managed Conda environment.
+"""
+
+import os
+import platform
+import requests
+import shutil
+import subprocess
+import tarfile
+import traceback
+from pathlib import Path, PurePosixPath
+from typing import Iterable, Tuple
+from urllib.parse import urljoin
+from ..types import RunnerSetupStatus, RunnerTestResults, RunnerUpdateStatus
+from ..util import capture_output, exec_or_return, user_app_data_dir, warn
+
+
+USER_APP_DATA_DIR = user_app_data_dir()
+
+PREFIX     = USER_APP_DATA_DIR / "runtimes/managed-conda"
+PREFIX_BIN = PREFIX / "bin"
+
+MICROMAMBA_ROOT = USER_APP_DATA_DIR / "micromamba"
+MICROMAMBA      = MICROMAMBA_ROOT / "bin/micromamba"
+
+
+def register_arguments(parser) -> None:
+    """
+    No-op.  No arguments necessary.
+    """
+    pass
+
+
+def run(opts, argv, working_volume = None, extra_env = {}, cpus: int = None, memory: int = None) -> int:
+    if working_volume:
+        os.chdir(str(working_volume.src))
+
+    # XXX TODO: In the future we might want to set rlimits based on cpus and
+    # memory, at least on POSIX systems.
+    #   -trs, 21 May 2020 (copied from ./native.py on 30 Aug 2022)
+
+    extra_env.update({
+        "PATH": path_with_prefix(extra_env),
+    })
+
+    return exec_or_return(argv, extra_env)
+
+
+def path_with_prefix(env = {}) -> str:
+    """
+    Prepends our prefix onto the existing value of ``PATH``, which is taken
+    from the first place it's found:
+
+      1. *env*
+      2. :py:attr:`os.environ`
+      3. :py:attr:`os.defpath`
+
+    Returns the modified ``PATH`` string.
+    """
+    # XXX TODO: Consider not inheriting the existing PATH, so as not to leak
+    # the user's ambient environment into this runtime.  In that case, we'd
+    # want PATH to be something like: our runtime prefix, standalone
+    # installation prefix (for `nextstrain`, if applicable), default path
+    # (os.defpath).
+    #   -trs, 13 Sept 2022
+    PATH = env.get("PATH", os.environ.get("PATH", os.defpath))
+    return os.pathsep.join([str(PREFIX_BIN), PATH])
```

Comment:
I drafted [an implementation of this TODO](https://gist.github.com/tsibley/3f055cce0911dab260edf1ee2852e057) as a configurable "isolation mode". I'd appreciate feedback on it, however, as I'm not sure if a) we want to default to the stricter or relaxed behaviour and b) if we should even allow switching from whatever mode we pick.

Reply:
My 2 cents: Allowing relaxed behavior would be a way to allow programs outside those installed by the (managed) runtime but I think this is going to be more trouble than it's worth and cause some hard-to-debug headaches. I would enforce the stricter behavior and if that's not enough for a given use-case then it's really not hard to create an ambient runtime!

Reply:
Thanks for the cents! I think that's my inclination as well, as folks with custom needs have several ways to add in other programs, including a) using the ambient runtime instead and b) modifying `PATH` themselves inside the workflow to make other programs available (even by, for example, using Snakemake's `--conda` support).